server/server.go

```diff
@@ -3,28 +3,62 @@ package server
 import (
 	"context"
 	"fmt"
+
 	pb "github.com/rode/rode/proto/v1alpha1"
-	grafeas "github.com/rode/rode/protodeps/grafeas/proto/v1beta1/grafeas_go_proto"
+	grafeas_proto "github.com/rode/rode/protodeps/grafeas/proto/v1beta1/grafeas_go_proto"
+	grafeas_project_proto "github.com/rode/rode/protodeps/grafeas/proto/v1beta1/project_go_proto"
 
 	"github.com/golang/protobuf/proto"
 	"go.uber.org/zap"
+	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 	"google.golang.org/protobuf/encoding/protojson"
 )
 
+// NewGrafeasClients construct for GrafeasClients
+func NewGrafeasClients(grafeasEndpoint string) (*GrafeasClients, error) {
+	connection, err := grpc.Dial(grafeasEndpoint, grpc.WithInsecure())
+	if err != nil {
+		return nil, err
+	}
+
+	grafeasClient := grafeas_proto.NewGrafeasV1Beta1Client(connection)
+	projectsClient := grafeas_project_proto.NewProjectsClient(connection)
+
+	return &GrafeasClients{grafeasClient, projectsClient}, err
+}
+
+// NewRodeServer constructor for rodeServer
+func NewRodeServer(logger *zap.Logger, grafeasClients GrafeasClients) (pb.RodeServer, error) {
+	rodeServer := &rodeServer{
+		logger:         logger,
+		GrafeasClients: grafeasClients,
+	}
+	if err := rodeServer.initialize(context.Background()); err != nil {
+		return nil, fmt.Errorf("failed to initialize rode server: %s", err)
+	}
+	return rodeServer, nil
+}
+
+// GrafeasClients container for Grafeas protobuf clients
+type GrafeasClients struct {
```

Comment:
making a custom type for this seems a bit strange to me - can we just attach the common and project clients to the `rodeServer` struct directly?

Reply:
this just reduces the number of arguments to return and pass into the server constructor

Reply:
I think I'd prefer if we got rid of it.  I'm not seeing a ton of value using this as an abstraction.  I think it's okay to be explicit with the arguments we pass to the constructor.

Reply:
👍